iroh/src/commands/sync.rs

```diff
@@ -354,3 +480,273 @@ async fn print_entry(doc: &Doc, entry: &Entry, content: bool) -> anyhow::Result<
     }
     Ok(())
 }
+
+fn canonicalize_path(path: &str) -> anyhow::Result<PathBuf> {
+    let path = PathBuf::from(shellexpand::tilde(&path).to_string());
+    Ok(path)
+}
+
+fn tag_from_file_name(path: &Path) -> anyhow::Result<Tag> {
+    match path.file_name() {
+        Some(name) => name
+            .to_os_string()
+            .into_string()
+            .map(|t| t.into())
+            .map_err(|e| anyhow!("{e:?} contains invalid Unicode")),
+        None => bail!("the given `path` does not have a proper directory or file name"),
+    }
+}
+
+pub async fn import_coordinator(
```

Comment:
It would be nice to have either some docs or a struct. Otherwise it is a bit hard to know what the two integers are that are returned.

I think we have (total_number, total_size) in a number of places, maybe we can make a struct in iroh-bytes...